src/targets/gpu/device/prefix_scan_sum.cpp

```diff
@@ -24,12 +51,32 @@ void prefix_scan_sum(hipStream_t stream, const argument& result, const argument&
                         k[axis] = j;
                         return k;
                     };
-                    block_scan<block_size>(idx,
-                                           sum{},
-                                           0,
-                                           n,
-                                           [&](auto j) { return input[compute_idx(j)]; },
-                                           [&](auto j, auto x) { output[compute_idx(j)] = x; });
+                    if(reverse)
```

Comment:
The `if` branches need to be moved outside of `gs_launch` or they need to use `if constexpr`. To use it in `if constexpr` you need to use a visitor:

```cpp
template<class F>
void visit_bool(bool b, F f)
{
    if (b)
        f(std::true_type{});
    else
        f(std::false_type{});
}
```

Then you can do something like:

```cpp
visit_bool(exclusive, [&](auto exclusive_c) {
    if constexpr(exclusive_c)
    {
        ...
    }
});
```